lib/identity_cache/cache_key_generation.rb

```diff
@@ -7,14 +7,17 @@ def self.schema_to_string(columns)
       columns.sort_by(&:name).map{|c| "#{c.name}:#{c.type}"}.join(',')
     end
 
-    def self.denormalized_schema_hash(klass)
+    def self.denormalized_schema_hash(klass, seen_assocs = Set.new)
       schema_string = schema_to_string(klass.columns)
       klass.send(:all_cached_associations).sort.each do |name, options|
         klass.send(:check_association_scope, name)
         ParentModelExpiration.check_association(options) if options[:embed]
         case options[:embed]
         when true
-          schema_string << ",#{name}:(#{denormalized_schema_hash(options[:association_reflection].klass)})"
+          unless seen_assocs.add?(name)
+              return IdentityCache.memcache_hash(schema_to_string(klass.columns))
```

Comment:
No, that's the intended behaviour. When we see a model that we've already encountered, that model's associations have already put into the cache higher up in the call stack, so there's no need to put them in the cache key again.